sonar-groovy-plugin/src/main/java/org/sonar/plugins/groovy/foundation/GroovyHighlighterAndTokenizer.java

```diff
@@ -0,0 +1,253 @@
+    GroovyLexer.LITERAL_int,
+    GroovyLexer.LITERAL_interface,
+    GroovyLexer.LITERAL_long,
+    GroovyLexer.LITERAL_native,
+    GroovyLexer.LITERAL_new,
+    GroovyLexer.LITERAL_null,
+    GroovyLexer.LITERAL_package,
+    GroovyLexer.LITERAL_private,
+    GroovyLexer.LITERAL_protected,
+    GroovyLexer.LITERAL_public,
+    GroovyLexer.LITERAL_return,
+    GroovyLexer.LITERAL_short,
+    GroovyLexer.LITERAL_static,
+    GroovyLexer.LITERAL_super,
+    GroovyLexer.LITERAL_switch,
+    GroovyLexer.LITERAL_synchronized,
+    GroovyLexer.LITERAL_this,
+    GroovyLexer.LITERAL_threadsafe,
+    GroovyLexer.LITERAL_throw,
+    GroovyLexer.LITERAL_throws,
+    GroovyLexer.LITERAL_trait,
+    GroovyLexer.LITERAL_transient,
+    GroovyLexer.LITERAL_true,
+    GroovyLexer.LITERAL_try,
+    GroovyLexer.LITERAL_void,
+    GroovyLexer.LITERAL_volatile,
+    GroovyLexer.LITERAL_while
+  };
+
+  private static final int[] STRINGS = {
+     GroovyLexer.STRING_CH,
+     GroovyLexer.STRING_CONSTRUCTOR,
+     GroovyLexer.STRING_CTOR_END,
+     GroovyLexer.STRING_CTOR_MIDDLE,
+     GroovyLexer.STRING_CTOR_START,
+     GroovyLexer.STRING_LITERAL,
+     GroovyLexer.STRING_NL,
+     GroovyLexer.REGEX_MATCH,
+     GroovyLexer.REGEX_FIND,
+     GroovyLexer.REGEXP_LITERAL,
+     GroovyLexer.REGEXP_CTOR_END,
+     GroovyLexer.REGEXP_SYMBOL,
+     GroovyLexer.DOLLAR_REGEXP_LITERAL,
+     GroovyLexer.DOLLAR_REGEXP_CTOR_END,
+     GroovyLexer.DOLLAR_REGEXP_SYMBOL,
+     GroovyLexer.DOLLAR,
+     GroovyLexer.ESCAPED_DOLLAR,
+     GroovyLexer.ESCAPED_SLASH
+  };
+
+  private static final int[] CONSTANTS = {
+    GroovyLexer.DIGIT,
+    GroovyLexer.DIGITS_WITH_UNDERSCORE,
+    GroovyLexer.DIGITS_WITH_UNDERSCORE_OPT,
+    GroovyLexer.HEX_DIGIT,
+    GroovyLexer.NUM_BIG_DECIMAL,
+    GroovyLexer.NUM_BIG_INT,
+    GroovyLexer.NUM_DOUBLE,
+    GroovyLexer.NUM_FLOAT,
+    GroovyLexer.NUM_INT,
+    GroovyLexer.NUM_LONG
+  };
+
+  private static final int[] COMMENTS = {
+    GroovyLexer.ML_COMMENT,
+    GroovyLexer.SH_COMMENT,
+    GroovyLexer.SL_COMMENT
+  };
+
+  private static final List<TypeOfTextToTokenTypes> HIGHLIGHTING_MAPPING = Lists.newArrayList(
+    new TypeOfTextToTokenTypes(TypeOfText.KEYWORD, KEYWORDS),
+    new TypeOfTextToTokenTypes(TypeOfText.STRING, STRINGS),
+    new TypeOfTextToTokenTypes(TypeOfText.CONSTANT, CONSTANTS),
+    new TypeOfTextToTokenTypes(TypeOfText.COMMENT, COMMENTS));
+
+  private final InputFile inputFile;
+  private final File file;
+  private boolean isAnnotation;
+
+  public GroovyHighlighterAndTokenizer(InputFile inputFile) {
+    this.inputFile = inputFile;
+    this.file = inputFile.file();
+  }
+
+  public void processFile(SensorContext context) {
+    List<GroovyToken> tokens = new ArrayList<>();
+    isAnnotation = false;
+
+    try (InputStreamReader streamReader = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8)) {
+      GroovyLexer groovyLexer = new GroovyLexer(streamReader);
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "FileInputStream". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2095)